kenny-lab/app/src/main/java/com/korbkenny/kennylab/Helper.java

```diff
@@ -0,0 +1,177 @@
+        SQLiteDatabase db = getWritableDatabase();
+        ContentValues values = new ContentValues();
+        values.put(COL_SSN, employee.getSsn());
+        values.put(COL_NAME_FIRST, employee.getFirstName());
+        values.put(COL_NAME_LAST, employee.getLastName());
+        values.put(COL_BIRTH_YEAR, employee.getYear());
+        values.put(COL_CITY, employee.getCity());
+        db.insertOrThrow(EMPLOYEE_TABLE_NAME,null,values);
+    }
+
+    public void insertRowJob(Job job){
+        SQLiteDatabase db = getWritableDatabase();
+        ContentValues values = new ContentValues();
+        values.put(COL_SSN, job.getSsn());
+        values.put(COL_COMPANY, job.getCompany());
+        values.put(COL_SALARY, job.getSalary());
+        values.put(COL_EXPERIENCE, job.getExperience());
+        db.insertOrThrow(JOB_TABLE_NAME,null,values);
+    }
+
+    private Helper(Context context){super(context,"db",null,1);}
+
+    private static Helper INSTANCE;
+
+    public static synchronized Helper getInstance(Context context){
+        if(INSTANCE == null)
+            INSTANCE = new Helper(context.getApplicationContext());
+        return INSTANCE;
+    }
+
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+        db.execSQL(CREATE_EMPLOYEE_TABLE);
+        db.execSQL(CREATE_JOB_TABLE);
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int i, int i1) {
+        db.execSQL(DELETE_TABLE_EMPLOYEE);
+        db.execSQL(DELETE_TABLE_JOB);
+        onCreate(db);
+    }
+
+
+    public List<String> getMacysWorkers(){
+        SQLiteDatabase db = getReadableDatabase();
+        String query = "SELECT " + COL_NAME_FIRST + ", " + COL_NAME_LAST + ", " +
+                COL_COMPANY + " FROM " + EMPLOYEE_TABLE_NAME + " INNER JOIN " +
+                JOB_TABLE_NAME + " ON " + EMPLOYEE_TABLE_NAME + "." + COL_SSN +
+                " = " + JOB_TABLE_NAME + "." + COL_SSN;
+
+        Cursor cursor = db.rawQuery(query,null);
+
+        List<String> names = new ArrayList<>();
+
+        if(cursor.moveToFirst()){
+            while(!cursor.isAfterLast()){
+                if((cursor.getString(cursor.getColumnIndex(COL_COMPANY))).equals("Macy's")) {
+                    String first = cursor.getString(cursor.getColumnIndex(COL_NAME_FIRST));
+                    String last = cursor.getString(cursor.getColumnIndex(COL_NAME_LAST));
+                    String name = first + " " + last;
+                    names.add(name);
+                }
+                cursor.moveToNext();
+            }
+        }
+        cursor.close();
+        return names;
+    }
+
+    public List<String> getCompaniesInBoston(){
+        SQLiteDatabase db = getReadableDatabase();
+        String query = "SELECT " + COL_COMPANY + ", " + COL_CITY + " FROM " +
+                EMPLOYEE_TABLE_NAME + " INNER JOIN " +  JOB_TABLE_NAME + " ON " +
+                EMPLOYEE_TABLE_NAME + "." + COL_SSN + " = " +
+                JOB_TABLE_NAME + "." + COL_SSN;
```

Comment:
As with the previous query, you could add the additional criteria as a where clause by putting `WHERE COL_CITY = 'BOSTON'` at the end. This is important because if your database is very large, getting ALL the matching rows then manually filtering out the ones you don't want can be extremely slow. It's better to let the database do the work of filtering by using a WHERE clause.